src/ui.js

```diff
@@ -1,169 +1,164 @@
-          buttons[types.POLYGON] = createControlButton({
-            container: controlGroup,
-            className: 'mapbox-gl-draw_polygon',
-            title: `Polygon tool ${ctx.options.keybindings && '(p)'}`,
-            onActivate: () => ctx.api.changeMode('draw_polygon')
-          });
-        }
+    if (controls[Constants.types.POINT]) {
+      buttonElements[Constants.types.POINT] = createControlButton(Constants.types.POINT, {
+        container: controlGroup,
+        className: Constants.CONTROL_BUTTON_POINT_CLASS,
+        title: `Marker tool ${ctx.options.keybindings && '(m)'}`,
+        onActivate: () => ctx.api.changeMode(Constants.modes.DRAW_POINT)
+      });
+    }
 
-        if (controls[types.POINT]) {
-          buttons[types.POINT] = createControlButton({
-            container: controlGroup,
-            className: 'mapbox-gl-draw_point',
-            title: `Marker tool ${ctx.options.keybindings && '(m)'}`,
-            onActivate: () => ctx.api.changeMode('draw_point')
-          });
+    if (controls.trash) {
+      buttonElements.trash = createControlButton('trash', {
+        container: controlGroup,
+        className: Constants.CONTROL_BUTTON_TRASH_CLASS,
+        title: 'Delete',
+        onActivate: () => {
+          ctx.api.trash();
         }
+      });
+    }
+  }
 
-        if (controls.trash) {
-          buttons.trash = createControlButton({
-            container: controlGroup,
-            className: 'mapbox-gl-draw_trash',
-            title: 'Delete',
-            onActivate: () => {
-              ctx.api.trash();
-              ctx.ui.setButtonInactive('trash');
-            }
-          });
-        }
-      },
-      setButtonActive: function(id) {
-        if (buttons[id] && id !== 'trash') {
-            buttons[id].classList.add('active');
-        }
-      },
-      setButtonInactive: function(id) {
-        if (buttons[id]) {
-          buttons[id].classList.remove('active');
-        }
-      },
-      setAllInactive: function() {
-        var buttonIds = Object.keys(buttons);
-
-        buttonIds.forEach(buttonId => {
-          if (buttonId !== 'trash') {
-            var button = buttons[buttonId];
-            button.classList.remove('active');
-          }
-        });
-      },
-      removeButtons: function() {
-        var buttonIds = Object.keys(buttons);
-
-        buttonIds.forEach(buttonId => {
-          var button = buttons[buttonId];
-          if (button.parentNode) {
-            button.parentNode.removeChild(button);
-          }
-          buttons[buttonId] = null;
-        });
+  function removeButtons() {
+    Object.keys(buttonElements).forEach(buttonId => {
+      const button = buttonElements[buttonId];
+      if (button.parentNode) {
+        button.parentNode.removeChild(button);
       }
-    };
+      delete buttonElements[buttonId];
+    });
+  }
+
+  return {
+    setButtonActive,
+    deactivateButtons,
+    queueContainerClasses,
```

Comment:
Container is a bit of an overloaded word in Draw. I think there are two containers right now (one for the full map and one for where the controls go). May `queueMapClasses` would be better?